microcloud/service/microceph.go

```diff
@@ -131,10 +133,33 @@ func (s CephService) Join(ctx context.Context, joinConfig JoinConfig) error {
 }
 
 // ClusterMembers returns a map of cluster member names and addresses.
-func (s CephService) ClusterMembers(ctx context.Context) (map[string]string, error) {
-	client, err := s.Client("", "")
-	if err != nil {
-		return nil, err
+// Optionally sends a remote request using the information in ServerInfo.
+func (s CephService) ClusterMembers(ctx context.Context, info *mdns.ServerInfo) (map[string]string, error) {
+	var client *client.Client
+	var err error
+	if info != nil && info.Name != s.name {
+		client, err = s.m.RemoteClient(util.CanonicalNetworkAddress(info.Address, CloudPort))
+		if err != nil {
+			return nil, err
+		}
+
+		client.Client.Client.Transport = &http.Transport{
```

Comment:
I'll have a PR coming soon that gets rid of these entirely, thanks to https://github.com/canonical/microcluster/pull/83

To keep this PR simple I've maintained the pre-existing behaviour. There's actually a few significant refactors that need to be made to the client generation in MicroCloud because we inefficiently open clients all over the place. I've been thinking of maintaining a map of clients in memory so that we don't need to constantly instantiate new ones throughout the init process.

Reply:
Ok, thanks for the details and the link. Fine for me then.